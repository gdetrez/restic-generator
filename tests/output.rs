use assert_cmd::prelude::*; // Add methods on commands
use std::{error::Error, fs::read_to_string, process::Command};
use tempfile::TempDir;

macro_rules! snapshot_test {
    ($name:ident, $config:expr, $output:expr) => {
        #[test]
        fn $name() -> Result<(), Box<dyn Error>> {
            let normal_dir = TempDir::new()?;
            let early_dir = TempDir::new()?;
            let late_dir = TempDir::new()?;
            let mut cmd = Command::cargo_bin("systemd-restic-generator")?;
            cmd.arg("-c")
                .arg($config)
                .arg(normal_dir.path())
                .arg(early_dir.path())
                .arg(late_dir.path());
            cmd.assert().success();
            insta::assert_snapshot!(read_to_string(normal_dir.path().join($output))?);
            Ok(())
        }
    };
}

snapshot_test!(
    local_backup_service,
    "example-config.toml",
    "restic-myrepo-backup.service"
);

snapshot_test!(
    remote_backup_service,
    "example-config.toml",
    "restic-sftprepo-backup.service"
);

snapshot_test!(
    s3_backup_service,
    "example-config.toml",
    "restic-s3bucket-backup.service"
);
