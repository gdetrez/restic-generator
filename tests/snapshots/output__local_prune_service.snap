---
source: tests/output.rs
expression: "read_to_string(normal_dir.path().join(\"restic-myrepo-prune.service\"))?"
---
# generated by restic-generator
[Unit]
Description=Prune /my/restic/repo
SourcePath=example-config.toml

[Service]
Environment=RESTIC_REPOSITORY="/my/restic/repo"
Environment=RESTIC_PASSWORD_COMMAND="pass restic/myrepo"
Type=oneshot
ExecStartPre=restic unlock
ExecStart=restic prune
Nice=10
IOSchedulingClass=idle

